src/classes/GS_AdminSetup.cls

```diff
@@ -0,0 +1,375 @@
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Get Started
+* @description Class to hold functionality to display Checklist in GS UI.
+*/
+public with sharing class GS_AdminSetup {
+
+    private static final String DISABLED = 'Disabled';
+    private static Set<Id> storedItemsId = null;
+    /**
+     * @description Returns a list of the checklist section with each subsection.
+     * Each section is defined in GS_Admin_Checklist_Section__mdt and each subsection is defined in GS_Admin_Checklist_Item__mdt
+     * @return List<ChecklistSection>
+     */
+    @AuraEnabled
+    public static List<ChecklistSection> getChecklists() {
+        storedItemsId = getMarkedAsCompletedCheckListItemByUser();
+        List<GS_Admin_Checklist_Section__mdt> checklistSections = getSectionsAndSubsections();
+        List<ChecklistSection> checklistWrapers = new List<ChecklistSection>();
+        for (GS_Admin_Checklist_Section__mdt checklistSection : checklistSections) {
+            checklistWrapers.add(wrapChecklistSection(checklistSection));
+        }
+        return checklistWrapers;
+    }
+
+    /**
+     * @description Persist in database if an element is checked or not.
+     * For doing so we use the object GetStartedCompletionChecklistState__c;. 
+     * The object contains two fields: Item_Id and User. 
+     * Which means that a checkbox is selected if an element  
+     * it means the element is checked
+     * @param  itemId  Id the checklist item id
+     * @param  isChecked boolean element is checked of not
+     */
+    @AuraEnabled
+    public static void updateCheckItem(String itemId, Boolean isChecked){
+               
+        if (doesChecklistItemExists(itemId)) {
+            List<GetStartedCompletionChecklistState__c> alreadyStoredCompletedItems = [SELECT Id
+                                                                                    FROM   GetStartedCompletionChecklistState__c
+                                                                                    WHERE  Item_Id__c = :itemId 
+                                                                                    AND    User__c = :UserInfo.getUserId()];
+            if (isChecked) {
+                markItemAsCompleted(alreadyStoredCompletedItems, itemId);
+            } else {
+                unmarkItemAsCompleted(alreadyStoredCompletedItems);
+            }
+        }    
+    }
+
+    /**
+     * @description Check if exists an GS_Admin_Checklist_Item__mdt with the Id
+     * @param  itemId String the id of the GS_Admin_Checklist_Item__mdt to be validated
+     * @return true if itemId is found, false otherwise
+     */
+    private static Boolean doesChecklistItemExists(String itemId) {
+        List<GS_Admin_Checklist_Item__mdt> items = [SELECT Id 
+                                                    FROM   GS_Admin_Checklist_Item__mdt
```

Comment:
Since it was stated that this page is meant for admins only, let's include the `WITH_SECURITY_ENFORCED` clause on this query. It'll be worth testing how the page renders for users without the appropriate permissions.

Reply:
I dont think it is the perfect timing to add changes in Apex code taking in consider we are in the last day with US team support this week, Yisel did the entire regression today. Doing this would mean running cautious QA process and I am afraid of lacking of time. I would do it as a separate WI without blocking this merge. I dont think doing could create a security issue taking in consider the page will only be rendered to Admins and the information that the query returns is less than meaningful

Reply:
Agreed. I do want to add these in and review security behavior. Can you ask Ali to add a new work item for security updates? It would consist of making sure that queries use the WITH SECURITY ENFORCED clause as well as verifying behavior when the User does not have proper permissions  to any of the objects or fields.